server/src/main/java/org/elasticsearch/common/io/stream/DelayableWriteable.java

```diff
@@ -46,7 +46,9 @@
      * when {@link #expand()} is called.
      */
     public static <T extends Writeable> DelayableWriteable<T> delayed(Writeable.Reader<T> reader, StreamInput in) throws IOException {
-        return new Serialized<>(reader, in.getVersion(), in.namedWriteableRegistry(), in.readBytesReference());
+        // TODO: the BytesReference read here can be very large => we should move to not copying the bytes by using
```

Comment:
Doing this is a little more involved since we have to cover all the corner cases to not leak buffers but well worth the effort eventually I think. I did this to provide some short-term relief (to G1GC users which is essentially everyone now).